pkg/server/storage/s3.go

```diff
@@ -0,0 +1,73 @@
+package storage
+
+import (
+	"io"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type S3Client interface {
+	PutObject(*s3.PutObjectInput) (*s3.PutObjectOutput, error)
+}
+
+type S3 struct {
+	Client           S3Client
+	Key              string
+	Secret           string
+	Region           string
+	Bucket           string
+	Endpoint         string
+	DisableSSL       bool
+	S3ForcePathStyle bool
+}
+
+func (s *S3) GetK8sSecretName() string {
+	return "s3-storage"
```

Comment:
Are you thinking to use k8s secrets to handle aws credentials ?
(to tell you true, I don't understand why we need secrets for s3 integration)